docs/docs/porting-create-react-app-to-gatsby.md

```diff
@@ -0,0 +1,267 @@
+  <Router>
+    <Route path="/user/" component={Users} />
+    <Route path="/user/:id" component={UserDetails} />
+  </Router>
+)
+
+export default App
+```
+
+Gatsby provides a `<Link />` component and a `navigate` function to help you direct users through pages on your site. You can read about how to use each in the [`gatsby-link` doc](/docs/gatsby-link/).
+
+### Handling state
+
+Because Gatsby rehydrates into a regular React app, state can be handled inside of components in the same way it would in Create React App. If you use a another library for state management and want to wrap your app in some sort of global state the section on [context providers](#context-providers) will be helpful.
+
+### Environment Variables
+
+Create React App requires you to create environment variables prefixed with `REACT_APP_`. Gatsby instead uses the `GATSBY_` prefix to [make environment variables accessible](/docs/environment-variables) in the browser context.
+
+```title=".env"
+# in Create React App
+REACT_APP_API_URL=http://someapi.com
+
+# in Gatsby
+GATSBY_API_URL=http://someapi.com
+```
+
+### Advanced customizations
+
+Part of Gatsby's philosophy around tooling is [progressively disclosing complexity](/docs/gatsby-core-philosophy/#progressively-disclose-complexity), this simplifies the experience for a wider audience while still allowing the option to configure more advanced features for those that feel inclined. You won't have to "eject" your Gatsby app to edit more complex configurations.
+
+In terms of levels of abstraction, Gatsby allows you to move up or down to tap into more sophisticated, and lower-level APIs without needing to eject like you would in Create React App.
+
+#### webpack
+
+Create React App will require you to eject or rely on another workaround to edit the webpack configuration. Gatsby allows [custom configuration of webpack](/docs/add-custom-webpack-config/) via the `gatsby-node.js` file.
+
+#### Context providers
+
+React's context API allows you to share state from a higher component and distribute it to components below it in the component tree without having to deal with issues like prop drilling (link?)
+
+How do you share state across components like a theme without one top level `App.js` file? Gatsby has a wrapRootElement and a wrapPageElement API that allow you to wrap the root element or all pages of your Gatsby site with components you want.
+
+In Create React App it could look like this:
+
+```jsx:title="create-react-app/src/App.js"
+import React from "react"
+
+const defaultTheme = "light"
+const ThemeContext = React.createContext(defaultTheme)
+
+function App() {
+  return (
+    <ThemeContext.Provider value={defaultTheme}>
+      {/* App, routing, and other components */}
+    </ThemeContext.Provider>
+  )
+}
+
+export default App
+```
+
+In Gatsby, if you want your providers to be global across pages you would move those providers to `gatsby-browser.js`:
+
+```jsx:title="gatsby/gatsby-browser.js"
```

Comment:
```suggestion
```jsx:title=gatsby/gatsby-browser.js
```

Reply:
Oh and for the _why_ of this, quotes aren't stripped and the title is treated as literal!